event/event.go

```diff
@@ -0,0 +1,13 @@
+package event
+
+import (
+	"github.com/ipld/go-ipld-prime"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+type Event struct {
```

Comment:
do you think we also need a type for each event? for examlpe to easily parse events with type: "pin", or "join"

Reply:
I have kept the event abstract on purpose to discuss the decisions with yourself and @mehdibalouchi. 

Here's a quick summary on how it occurs for me: we probably need types for events yep. This is more to facilitate fluent way to work with IPLD nodes. On how it gets encoded, one can imagine a flat chain where different types are linked together. Another I thought of was chained Event types as the example shows but with an additional link to specific event content. That could have advantages during traversal.

Reply:
I agree, that having the event content separated from the event head will have benefits when we need to query the chain and also for creating different views of the chain. But for this to happen, we have to keep a `Type` for each event descriptor (or head, the node that is part of the chain)

Reply:
@masih  I just saw in the test file what you mean, brilliant :D